nautobot/extras/plugins/views.py

```diff
@@ -30,24 +30,25 @@ class InstalledPluginsView(AdminRequiredMixin, View):
     table = InstalledPluginsTable
 
     def get(self, request):
-        plugins = [apps.get_app_config(plugin) for plugin in settings.PLUGINS]
         data = []
-        for plugin in plugins:
-            data.append(
-                {
-                    "name": plugin.verbose_name,
-                    "package_name": plugin.name,
-                    "author": plugin.author,
-                    "author_email": plugin.author_email,
-                    "description": plugin.description,
-                    "version": plugin.version,
-                    "actions": {
-                        "home": plugin.home_view_name,
-                        "configure": plugin.config_view_name,
-                        "docs": plugin.docs_view_name,
-                    },
-                }
-            )
+        for plugin in apps.get_app_configs():
+            if plugin.name in settings.PLUGINS:
```

Comment:
I wonder if we should also add a `and issubclass(plugin, NautobotAppConfig)` as protection against errors, since most of the below attributes aren't present in the base `AppConfig` class?

Reply:
All apps should be in the plugins list and all apps should have the config a subclass of either `NautobotAppConfig` or `nautobot.extras.plugins.PluginConfig` (later is a subclass of the former with a deprecation warning). Not opposed to adding what value does it have for the view? I would think the startup should error out if an app isn't a subclass of one of the above ancestors right?

Reply:
I haven't actually tried to see what happens if you have something in PLUGINS that's using AppConfig instead of NautobotAppConfig. :-) You may well be correct that it's not needed at this point in the code.

Reply:
@glennmatthews Just checked, so _technically_ there's no validation on the ancestor BUT it requires the following attrs that aren't part of the the base `AppConfig` so it throws an `PluginImproperlyConfigured` exception on startup. I think that there's merit in doing validation before [this line](https://github.com/nautobot/nautobot/blob/512b1ee9062638ea726173abc910fc6bbc68bcfb/nautobot/extras/plugins/utils.py#L74)

* `config`
* `min_version`
* `max_version`
* `default_settings`
* `installed_apps`
* `middleware`
* `required_settings`

Reply:
Thanks for checking! Let's get an issue in the backlog to improve that validation; definitely doesn't need to be part of this PR.

Reply:
@glennmatthews I opened #4880